docs/clustering/downscaling.rst

```diff
@@ -0,0 +1,360 @@
+::
+
+   cluster.name: vanilla
+   node.name: <NODE_NAME>
+   network.host: _local_
+   node.max_local_storage_nodes: 3
+   stats.service.interval: 0
+
+   http.cors.enabled: true
+   http.cors.allow-origin: "*"
+
+   transport.tcp.port: <PORT>
+   gateway.expected_nodes: 3
+   gateway.recover_after_nodes: 3
+   discovery.seed_hosts:
+       - 127.0.0.1:4301
+       - 127.0.0.1:4302
+       - 127.0.0.1:4303
+   cluster.initial_master_nodes:
+       - 127.0.0.1:4301
+       - 127.0.0.1:4302
+       - 127.0.0.1:4303
+
+where ``<PORT>`` is one of ``[4301, 4302, 4303]``, and ``<NODE_NAME>`` one of ``[n1, n1, n3]``.
+
+The settings come explained in cluster-wide-settings_ and node-specific-settings_.
+
+
+Installing from sources
+=======================
+
+An alternative way of installing CrateDB is from its sources. This script summarizes
+the procedure:
+
+::
+
+    #!/bin/sh
+
+    # Forder 'dist' will contain the CrateDB git clone 'crate-clone'.
+    # The clone will be used to build a tarball, which will be
+    # cracked open. A soft link `crate` will be created to point
+    # to the CrateDB distribution's root folder.
+
+    if [ ! -d dist ]; then
+        mkdir dist
+    fi
+
+    # get the clone
+    if [ ! -d dist/crate-clone ]; then
+        git clone https://github.com/crate/crate.git dist/crate-clone
+    fi
+    cd dist/crate-clone
+
+    # delete any old tarballs
+    find ./app/build/distributions -name 'crate-*.tar.gz' -exec rm -f {} \;
+
+    # build the tarball
+    git pull
+    ./gradlew clean disTar
+
+    # crack open the tarball
+    latest_tar_ball=$(find ./app/build/distributions -name 'crate-*.tar.gz')
+    cp $latest_tar_ball ..
+    cd ..
+    name=$(basename $latest_tar_ball)
+    tar -xzvf $name
+    rm -f $name
+    name=${name/.tar.gz/}
+    cd ..
+
+    # create the symlink to the distribution folder
+    rm -f crate
+    ln -s dist/$name crate
+    echo "Crate $name has been installed."
+
+
+Once CrateDB is installed, you can create a folder ``conf`` at the same level as
+the ``crate`` soft link. This folder should contain a subfolder per node, named
+the same as the node ``[n1, n2, n3]``, where you can place the ``crate.yml`` file,
+as well as the ``log4j2.properties`` logger configuration file (you will find a
+copy inside the ``config`` folder within the CrateDB installation).
+
+To start each node ``[n1, n2, n3]`` you can run a script (once per node) similar to:
+
+::
+
+    node_name=$1
+
+    path_conf="$(pwd)/conf/$node_name"
+    path_home=$(pwd)/crate
+    path_data=$(pwd)/data/$node_name
+    path_snapshots=$(pwd)/data/$node_name/snapshots
+
+    export CRATE_HOME=$path_home
+    if [ -z "$CRATE_HEAP_SIZE" ]; then
+        export CRATE_HEAP_SIZE="8G"
+    fi
+
+    ./crate/bin/crate -Cpath.conf=$path_conf \
+                      -Cpath.data=$path_data \
+                      -Cpath.repo=$path_snapshots \
+                      -Cnode.name=<NODE_NAME> \
+                      -Ctransport.tcp.port=<PORT>
```

Comment:
both settings are defined inside the `crate.yml` template above and thus not needed.